### ISSUE TITLE
Build does not work with the required CMake version

### ISSUE THREAD

**Issue by justsz** [Nicholaswogan/fortran-yaml-c#2]
Fetching dependencies during the build fails with the output below. It appears to be due to a CMake bug. Version 3.16.3 gives the error, but version 3.24.1 works fine. Not sure about versions between those two.

I suppose that `cmake_minimum_required(VERSION "3.14")` needs to be updated.

```
Setting up GNU base environment
Cloning into 'fyaml'...
remote: Enumerating objects: 175, done.
remote: Counting objects: 100% (175/175), done.
remote: Compressing objects: 100% (115/115), done.
remote: Total 175 (delta 100), reused 124 (delta 49), pack-reused 0
Receiving objects: 100% (175/175), 64.97 KiB | 2.24 MiB/s, done.
Resolving deltas: 100% (100/100), done.
-- The C compiler identification is GNU 9.4.0
-- The Fortran compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working Fortran compiler: /usr/bin/gfortran
-- Check for working Fortran compiler: /usr/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/gfortran supports Fortran 90
-- Checking whether /usr/bin/gfortran supports Fortran 90 -- yes
-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
-- Verifying Fortran/C Compiler Compatibility
-- Verifying Fortran/C Compiler Compatibility - Success
-- Downloading CPM.cmake to /home/bla/ci_vm_setup/fyaml/build/cmake/CPM_0.34.3.cmake
-- CPM: adding package libyaml@0.2.5 (0.2.5)
Scanning dependencies of target libyaml-populate
[ 11%] Creating directories for 'libyaml-populate'
[ 22%] Performing download step (git clone) for 'libyaml-populate'
Cloning into 'libyaml-src'...
Note: switching to '0.2.5'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 2c891fc Changes for v0.2.5 release
[ 33%] No patch step for 'libyaml-populate'
[ 44%] Performing update step for 'libyaml-populate'
fatal: invalid upstream 'origin/0.2.5'
fatal: No rebase in progress?
CMake Error at /home/bla/ci_vm_setup/fyaml/build/_deps/libyaml-subbuild/libyaml-populate-prefix/tmp/libyaml-populate-gitupdate.cmake:102 (message):


  Failed to rebase in:
  '/home/bla/ci_vm_setup/fyaml/build/_deps/libyaml-src/'.

  You will have to resolve the conflicts manually


make[2]: *** [CMakeFiles/libyaml-populate.dir/build.make:97: libyaml-populate-prefix/src/libyaml-populate-stamp/libyaml-populate-update] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/libyaml-populate.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

CMake Error at /usr/share/cmake-3.16/Modules/FetchContent.cmake:915 (message):
  Build step for libyaml failed: 2
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FetchContent.cmake:1006 (__FetchContent_directPopulate)
  build/cmake/CPM_0.34.3.cmake:873 (FetchContent_Populate)
  build/cmake/CPM_0.34.3.cmake:679 (cpm_fetch_package)
  CMakeLists.txt:12 (CPMAddPackage)


-- Configuring incomplete, errors occurred!
```

**Comment by Nicholaswogan** [comment]
Thanks for the issue. This is confusing because [CPM](https://github.com/cpm-cmake/CPM.cmake) states that CMake>=3.14 is required.

Can anything else explain why the build fails?

**Comment by justsz** [comment]
Ah, good point. I added `CPMAddPackage("gh:catchorg/Catch2@2.5.0")` just as a test, and that dependency was retrieved with CMake 3.16.3. Maybe it's something specific with libyaml.

**Comment by Nicholaswogan** [comment]
libyaml requires CMake version 3.0

**Comment by justsz** [comment]
I tried on a different system with cmake 3.18.2 and got the same problem.

The issue is triggered when cmake/cpm enters the libyaml-src and tries to execute `rebase origin/0.2.5`, which gives an `invalid upstream` error. Either `rebase 0.2.5` or `rebase origin/release/0.2.5` would work successfully. The older CMake generates `rebase ${remote}/${tag}` (==origin/0.2.5), while the newer one does `rebase ${checkout_name}` (== 0.2.5).

Setting `GIT_TAG "release/0.2.5"` in the fortran-yaml-c CMakeList.txt file appears to fix the issue, it works with both cmake versions.

**Comment by Nicholaswogan** [comment]
I am unable to install cmake v3.16.3. Can you try the following replacement cmake file to see if it works? Also, if you want to submit a pull request that is cool too

```cmake
cmake_minimum_required(VERSION "3.14")
project (FORTRAN_YAML_C VERSION "0.2.2" LANGUAGES C Fortran)

include(FortranCInterface)
FortranCInterface_VERIFY()

set(CMAKE_Fortran_MODULE_DIRECTORY "${CMAKE_BINARY_DIR}/modules")

# dependencies
include(cmake/CPM.cmake)
CPMAddPackage(
  NAME libyaml 
  VERSION 0.2.5
  GITHUB_REPOSITORY "yaml/libyaml"
  GIT_TAG "0.2.5"
  OPTIONS
    "GIT_TAG release/0.2.5"
  EXCLUDE_FROM_ALL ON
)

add_subdirectory(src)
add_subdirectory(tests)

```

**Comment by justsz** [comment]
Adding `options` did not fix it.

**Comment by Nicholaswogan** [comment]
Closed with 8407875e1775c08a9374b425aa7e7018470a331d
